src/PageObject/Element/Toolbar.php

```diff
@@ -0,0 +1,166 @@
+                $is_link = strtolower($link_text) === $current_item_name;
+        endswitch;
+
+        return $is_link;
+    }
+
+    /**
+     * Retruns a second-level toolbar NodeElement corresponding to the link text
+     * under the given first-level toolbar NodeElement.
+     *
+     * @param NodeElement $first_level_item The element to check under
+     * @param string $link_text The link (text) to check for
+     * @return NodeElement|null
+     */
+    protected function getSubmenuLinkNode($first_level_item, $link_text)
+    {
+        $second_level_items = $first_level_item->findAll('css', 'ul li a');
+        $submenu_link_node = null;
+        foreach ($second_level_items as $second_level_item) {
+            $current_item_name = Util\stripTagsAndContent($second_level_item->getHtml());
+            if (strtolower($link_text) === strtolower($current_item_name)) {
+                try {
+                    // "Focus" (add hover class) on the toolbar link so the submenu appears
+                    $id = $first_level_item->getAttribute('id');
+                    $this->getDriver()->evaluateScript(
+                        'jQuery( "#' . $id . '" ).addClass("hover");'
+                    );
+                } catch (UnsupportedDriverActionException $e) {
+                    // This will fail for GoutteDriver but neither is it necessary
+                }
+                $submenu_link_node = $second_level_item;
+                break;
+            }
+        }
+        return $submenu_link_node;
+    }
+
+    /**
+     * Searches for the given text using the toolbar search field.
+     *
+     * @param string $text Text to enter into the search field
+     */
+    public function search($text)
+    {
+        $search = $this->find('css', '#adminbarsearch');
+
+        if (! $search) {
+            throw new ExpectationException(
+                'Search field in the toolbar could not be found',
+                $this->getDriver()
+            );
+        }
+
+        try {
+            $search->find('css', '#adminbar-search')->press();
+            $search->find('css', '#adminbar-search')->focus();
+            $search->fillField('adminbar-search', $text);
+            $search->find('css', '#adminbar-search')->focus();
+        } catch (UnsupportedDriverActionException $e) {
+            // This will fail for GoutteDriver but neither is it necessary
+            $search->fillField('adminbar-search', $text);
+        }
+
+        $search->submit();
+    }
+
+    /**
+     * Logs-out using the toolbar log-out link.
+     */
+    public function logOut()
+    {
+        // Using NodeElement::mouseOver() won't work because WordPress is using hoverIndent. Instead we just
+        // manually add the hover class. See https://github.com/paulgibbs/behat-wordpress-extension/issues/65
+        try {
+            $this->getDriver()->evaluateScript(
+                'jQuery( "#wp-admin-bar-my-account" ).addClass("hover");'
```

Comment:
Doesn't really matter here but let's have the spacing around the brackets consistent! (I don't care strongly either way!)